### PR TITLE
feature: empty datasets

### PR DESCRIPTION
**Pull request by aaryanpunia**

### CODE REVIEW

File: server/src/bin/delete-worker.rs [hunk 1]
```diff
@@ -210,6 +210,28 @@ async fn delete_worker(
...
+                    log::info!(
+                        "Deleted all chunks for dataset: {:?}",
+                        delete_worker_message.dataset_id
+                    );
+                    continue;
+                }
+                Err(err) => {
+                    log::error!("Failed to delete all chunks for dataset: {:?}", err);
+                    continue;
+                }
```
Comment (skeptrunedev) [inline server/src/bin/delete-worker.rs:231]: If the response is an `Ok` then it's important to remove it from the processing queue like: 

```
let _ = redis::cmd("LREM")
                    .arg("delete_dataset_processing")
                    .arg(1)
                    .arg(serialized_message)
                    .query_async::<redis::aio::MultiplexedConnection, usize>(&mut *redis_connection)
                    .await;
```

If the response is an error, then you need to add it back to the queue: 

```
let _ = readd_error_to_queue(delete_worker_message, err, redis_pool.clone()).await;
```

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

File: server/src/bin/delete-worker.rs (hunk 1 above)
Comment (skeptrunedev) [reply server/src/bin/delete-worker.rs:231]: Can reference the code slightly lower than this for how this should work.

File: server/src/operators/dataset_operator.rs [hunk 2]
```diff
@@ -256,6 +288,12 @@ pub async fn delete_dataset_by_id_query(
             break;
         }
 
+        let old_count = chunk_metadata_columns::chunk_metadata
+            .count()
+            .get_result::<i64>(&mut conn)
+            .await
+            .unwrap();
+
```
Comment (skeptrunedev) [inline server/src/operators/dataset_operator.rs:296]: I think this and the following count should be removed?

**Review by skeptrunedev — CHANGES_REQUESTED** [review]

File: server/src/operators/dataset_operator.rs (hunk 2 above)
Comment (aaryanpunia) [reply server/src/operators/dataset_operator.rs:296]: ah, residue from the logging I was doing!

File: server/src/operators/dataset_operator.rs [hunk 3]
```diff
@@ -190,6 +191,37 @@ pub async fn soft_delete_dataset_by_id_query(
...
+    let mut redis_conn = redis_pool
+        .get()
+        .await
+        .map_err(|err| ServiceError::BadRequest(err.to_string()))?;
+
+    let message = DeleteMessage {
+        dataset_id: id,
+        server_config: config,
+        attempt_number: 0,
+        empty_database: true,
```
Comment (skeptrunedev) [inline server/src/operators/dataset_operator.rs:224]:
```suggestion
        empty_dataset: true,
```

Nit

**Review by skeptrunedev — APPROVED** [review]

**Pull request merged**